providers/oidc.go

```diff
@@ -196,6 +199,16 @@ func (p *OIDCProvider) CreateSessionFromToken(ctx context.Context, token string)
 	return ss, nil
 }
 
+// CreateSessionFromIntrospectedToken converts Bearer Tokens into sessions after valified using introspection endpoint
+func (p *OIDCProvider) CreateSessionFromIntrospectedToken(ctx context.Context, token string) (*sessions.SessionState, error) {
+
+	_, err := p.introspectToken(token)
+	if err != nil {
+		return nil, err
+	}
+	return p.CreateSessionFromToken(ctx, token)
```

Comment:
I just tested this branch in one of my environments and I identified an issue we will need to fix:

Access tokens are allowed to be random strings by the oauth2 specification. The current implementation presented here expects the access token to be a JWT token. Instead we need to implement new logic to construct a session with the information retrieved from the introspection endpoint.

Reply:
Indeed, even trying start a non-JWT proxy fails:
```
ERROR: Failed to initialise OAuth2 Proxy: error initialising provider: could not create provider data: error building OIDC ProviderVerifier: invalid provider verifier options: missing required setting: jwks-url
```

Reply:
Indeed, we also have to allow not specifying JWKS URL, beacause currrently it results in a fatal error:
```dosini
provider="oidc"
skip_oidc_discovery="true"
# jwks_url="not available"
```
```
[2023/10/27 12:11:43] [main.go:60] ERROR: Failed to initialise OAuth2 Proxy: error initialising provider: could not create provider data: error building OIDC ProviderVerifier: invalid provider verifier options: missing required setting: jwks-url
```

Reply:
that is expected behavior. if you disable discovery you will have to manually define a jwks url. ID Tokens are per specification jwt tokens.